test/instrument/control/if.t

```diff
@@ -97,3 +97,116 @@ tail position iff the whole if-expression is in tail position.
+  >   print_endline "this";
+  >   print_endline "that";
+  > EOF
+  let _ =
+    if bool_of_string "true" then (
+      ___bisect_visit___ 5;
+      ___bisect_post_visit___ 2 (print_endline "foo"))
+    else (
+      ___bisect_visit___ 4;
+      ___bisect_post_visit___ 3 (print_endline "bar"));
+    ___bisect_post_visit___ 1 (print_endline "this");
+    ___bisect_post_visit___ 0 (print_endline "that")
+
+Where there's a raise or a failwith guarded by a conditional, the
+branch itself is instrumented, but the raising expression is not
+post-instrumented...
+
+  $ bash ../test.sh <<'EOF'
+  > let _f ~cond1 ~cond2 =
+  >   if cond1 then failwith "this";
+  >   if cond2 then raise Not_found;
+  >   print_endline "this";
+  >   print_endline "that";
+  > EOF
+  let _f ~cond1 ~cond2 =
+    ___bisect_visit___ 5;
+    if cond1 then (
+      ___bisect_visit___ 4;
+      failwith "this");
+    ___bisect_visit___ 3;
+    if cond2 then (
+      ___bisect_visit___ 2;
+      raise Not_found);
+    ___bisect_visit___ 1;
+    ___bisect_post_visit___ 0 (print_endline "this");
+    print_endline "that"
+
+... however note that this doesn't hold for all raising expressions.
+For example, if you use `invalid_arg`, or a raising helper, its
+application will be post-instrumented (unless all of its arguments are
+labeled !?), resulting in a code location that can never be visited,
+by design. These are currently acknowledged limitations.
+
+  $ bash ../test.sh <<'EOF'
+  > let invalid_n_labeled ~n = invalid_arg (Printf.sprintf "Invalid value of n=%d" n)
+  > 
+  > let invalid_p_unlabeled p = invalid_arg (Printf.sprintf "Invalid value of p=%d" p)
+  > 
+  > let _f ~cond1 ~n ~p =
+  >   if cond1 then invalid_arg "this";
+  >   if n < 0 then invalid_n_labeled ~n;
+  >   if p < 0 then invalid_p_unlabeled p;
+  >   print_endline "this";
+  >   print_endline "that";
+  > EOF
+  let invalid_n_labeled ~n =
+    ___bisect_visit___ 1;
+    invalid_arg
+      (___bisect_post_visit___ 0 (Printf.sprintf "Invalid value of n=%d" n))
+  
+  let invalid_p_unlabeled p =
+    ___bisect_visit___ 3;
+    invalid_arg
+      (___bisect_post_visit___ 2 (Printf.sprintf "Invalid value of p=%d" p))
+  
+  let _f ~cond1 ~n ~p =
+    ___bisect_visit___ 13;
+    if cond1 then (
+      ___bisect_visit___ 12;
+      ___bisect_post_visit___ 11 (invalid_arg "this"));
+    ___bisect_visit___ 10;
+    if n < 0 then (
+      ___bisect_visit___ 9;
+      invalid_n_labeled ~n);
+    ___bisect_visit___ 8;
+    if p < 0 then (
+      ___bisect_visit___ 7;
+      ___bisect_post_visit___ 6 (invalid_p_unlabeled p));
+    ___bisect_visit___ 5;
+    ___bisect_post_visit___ 4 (print_endline "this");
+    print_endline "that"
+
+If you try and disable post-instrumentation with a `coverage off`
+directive, the branch pre-instrumentation is removed as well,
+resulting in losing the ability to control that the branch itself is
+visited.
```

Comment:
Could you move this to [`attribute.t`](https://github.com/aantron/bisect_ppx/blob/e30265643e77bcf2c9eba7322429c779122106fc/test/instrument/attribute.t)? That is the place for testing `[@coverage off]`. It already includes cases for interaction of `[@coverage off]` with multiple language constructs in case.

Reply:
I appreciate your suggestion to add tests to `attribute.t`, which I had
overlooked. I will proceed to incorporate new tests there to demonstrate how to
disable coverage in branches (same PR ok?).

Despite the presence of branch coverage testing in `attribute.t`, I advocate for
retaining this specific part of the control test here. It serves to illustrate
that this method does not provide a viable workaround for the issue at hand.
This was in response to your comment in #433:

> I believe in your case you should suppress instrumentation of the out-edge
> with [@coverage off], though I don't immediately recall whether the way that
> is implemented will also cause the branch instrumentation to also be
> supressed.

This might come across more as literate testing (akin to a tutorial, perhaps?)
rather than a pure behavior test. I've attempted to construct a small narrative
around the added tests. However, if you have strong reservations, I'm open to
removing this section entirely.